trading_calendars/exchange_calendar_xwar.py

```diff
@@ -12,92 +14,139 @@
 from pandas.tseries.holiday import (
     Holiday,
     GoodFriday,
-    EasterMonday
+    EasterMonday,
 )
-from pytz import timezone
+from pytz import timezone, UTC
+
 from .common_holidays import (
     new_years_day,
+    epiphany,
     european_labour_day,
     corpus_christi,
     all_saints_day,
-    boxing_day,
-    christmas,
     christmas_eve,
-    new_years_eve
-)
-from .trading_calendar import (
-    TradingCalendar,
-    HolidayCalendar
+    christmas,
+    boxing_day,
+    new_years_eve,
 )
+from .trading_calendar import HolidayCalendar, TradingCalendar
+
+
+def not_2004(datetime_index):
+    """
+    Christmas Eve is a holiday every year except for whatever reason it was a
+    trading day in 2004.
+    """
+    return datetime_index[datetime_index.year != 2004]
+
+
+NewYearsDay = new_years_day()
+
+Epiphany = epiphany(start_date='2011')
+
+LabourDay = european_labour_day()
 
 May3ConstitutionDay = Holiday(
-    "Celabration of declaration of the Constitution of 3 May",
+    "Celabration of Declaration of the Constitution of 3 May",
     month=5,
-    day=3
+    day=3,
 )
+
+CorpusChristi = corpus_christi()
+
 ArmedForcesDay = Holiday(
     "Armed Forces Day",
     month=8,
-    day=15
+    day=15,
 )
+
+AllSaintsDay = all_saints_day()
+
 IndependenceDay = Holiday(
     "National Independence Day",
     month=11,
-    day=11
+    day=11,
 )
 
+ChristmasEve = christmas_eve(observance=not_2004)
+Christmas = christmas()
+BoxingDay = boxing_day()
+
+NewYearsEve = new_years_eve(start_date='2011')
+
 
 class XWARExchangeCalendar(TradingCalendar):
     """
-        Exchange calendar for the Warsaw Stock Exchange (WSE).
-
-        Open Time: 9:00 AM, CET
-        Close Time: 5:30 PM, CET
-
-        Regularly-Observed Holidays:
-        - New Years Day
-        - Good Friday
-        - Easter Monday
-        - Labour Day
-        - May 3 Constitution Day
-        - Corpus Christi
-        - Armed Forces Day
-        - All Saints Day
-        - Independence Day
-        - Christmas Eve
-        - Christmas Day
-        - Boxing Day
-        - New Years Eve
-        """
+    Exchange calendar for the Warsaw Stock Exchange (WSE).
+
+    Open Time: 9:00 AM, Central European Time (CET)
+    Close Time: 5:00 PM, Central European Time (CET)
+
+    Regularly-Observed Holidays:
+      - New Year's Day
+      - Epiphany
+      - Good Friday
+      - Easter Monday
+      - Labour Day
+      - Constitution Day
+      - Corpus Christi
+      - Armed Forces Day
+      - All Saints' Day
+      - Independence Day
+      - Christmas Eve
+      - Christmas Day
+      - Boxing Day
+      - New Year's Eve
+
+    Holidays No Longer Observed:
+      - None
+
+    Early Closes:
+      - None
+    """
     name = 'XWAR'
 
-    tz = timezone('CET')
+    tz = timezone('Europe/Warsaw')
 
     open_times = (
-        (None, time(8, 30)),
+        (None, time(9, 1)),
     )
-
     close_times = (
-        (None, time(17, 5)),
+        (None, time(17)),
```

Comment:
gpw.pl/session-details has slightly different hours, not sure how we want to model that.

Reply:
@jmccorriston Do you have thoughts on this? I'm going to merge for now, and if we decide to revisit this we'll need to also address some of the other calendars as well.